lib/sgtn-client/util/cache-util.rb

```diff
@@ -35,16 +38,19 @@ def self.write_cache(cache_key, items)
       end
 
       def self.get_cachekey(component, locale)
+        if locale.nil?
+            return nil
+        end
         env = SgtnClient::Config.default_environment
         product_name = SgtnClient::Config.configurations[env]["product_name"]
         version = SgtnClient::Config.configurations[env]["version"].to_s
-        default_l = SgtnClient::Config.configurations[env]["default_language"]
+        default_l = SgtnClient::Config.configurations[env]["source_locale"]
         if default_l == nil
           default_l = 'en'
         end
         lc = locale == default_l ? SgtnClient::Config.configurations.default: locale
         SgtnClient.logger.debug "[CacheUtil]get cache key: #{lc}"
-        return product_name + "_" + version + "_" + component + "_" + lc
+        return product_name + "_" + version + "_" + component + "_" + lc.to_s
```

Comment:
This is for a different issue: https://github.com/vmware/singleton/issues/1479
Please move to a different PR.